cinema-backend/src/main/kotlin/pl/theliver/cinemabackend/application/services/SeanceService.kt

```diff
@@ -22,10 +24,55 @@ class SeanceService(
 
     fun getSeancesFromMovie(id: String) = seanceRepository.getAllSeancesByMovieId(id)
 
+    fun getSeancesByScreeningRoomId(id: String) = seanceRepository.getAllSeancesByScreeningRoomId(id)
+
+    fun deleteSeanceById(id: String) {
+        placeRepository.getAllPlacesBySeanceId(id).map { placeRepository.deleteById(it.id) }
+        reservationRepository.getAllReservationsBySeanceId(id).map { reservationRepository.deleteById(it.id) }
+        seanceRepository.deleteById(id)
+    }
+
     fun getSeancesByMovieIdAndDictForRoom(id: String): Pair<List<Seance>, Map<String, String>> {
         return Pair(
-                seanceRepository.getAllSeancesByMovieId(id),
-                screeningRoomRepository.getAllScreeningRooms().map { it.id to it.name }.toMap()
+            seanceRepository.getAllSeancesByMovieId(id),
+            screeningRoomRepository.getAllScreeningRooms().map { it.id to it.name }.toMap()
         )
     }
+
+    fun createNewSeanceIfNotHoursConflict(seance: Seance): Seance? {
```

Comment:
Method too long, if you need prepare private data classes to return types of smaller methods.

Reply:
Consider extension functions

Reply:
Better prepare method to check condition and method to create

Reply:
This file must to be in git